src/context/mod.rs

```diff
@@ -188,6 +199,85 @@ impl<C: Blockchain> MatchstickInstanceContext<C> {
         Ok(())
     }
 
+    /// function logEntity(entity: string, id: string, showRelated: bool): void
+    pub fn log_entity(
+        &mut self,
+        _gas: &GasCounter,
+        entity_type_ptr: AscPtr<AscString>,
+        entity_id_ptr: AscPtr<AscString>,
+        show_related_ptr: AscPtr<bool>,
+    ) -> Result<(), HostExportError> {
+        let entity_type: String = asc_get(&self.wasm_ctx, entity_type_ptr, &GasCounter::new(), 0)?;
+        let entity_id: String = asc_get(&self.wasm_ctx, entity_id_ptr, &GasCounter::new(), 0)?;
+        let show_related: bool = bool::from(EnumPayload(show_related_ptr.to_payload()));
+
+        let mut log: HashMap<String, LogEntityValue> = HashMap::new();
+
+        // validates whether the provided entity exists in the schema file
+        let is_invalid_entity = SCHEMA
```

Comment:
Wouldn't it be better if we initialize the store with all existing entities similarly to the derived fields, and then here to just check if the key exists in the store?